### PR TITLE
thunder sqlgen: Add support for nil values in CopySlice

### PR DESCRIPTION
**Pull request by silverspace**
Fixing a panic in CopySlice when the input slice contains nil values:
panic: reflect: call of reflect.Value.Set on zero Value

Also adding a test to verify the behavior of CopySlice with nil values.

### CODE REVIEW

File: batch/batch_test.go [hunk 1]
```diff
@@ -48,34 +48,35 @@ func TestBasic(t *testing.T) {
...
-		allowedCounts map[int]bool
+		expectedCount int
 	}{
 		{
 			description:   "Expect sleep less than WaitDuration to result in a single call being made.",
-			interval:      500 * time.Microsecond,
-			sleepDuration: 100 * time.Microsecond,
-			maxDuration:   10 * time.Millisecond,
-			allowedCounts: map[int]bool{1: true},
+			interval:      2 * sleepDuration,
```
Comment (silverspace) [inline batch/batch_test.go]: Sure, I'll file a follow-up PR to increase the default wait interval to 1ms. I'm reluctant to go too much higher since any request that isn't batched will pay this penalty for every batch invocation.

I'll also include metrics in the new PR for recording the cardinality of the batch request to DataDog, so we can see how this increases or decreases as we play with the DefaultWaitDuration.
